### PR TITLE
feature: add posts endpoint

### PR DESCRIPTION
**Pull request by CristopherH11**
# New Feature: Create Posts via Authenticated API Endpoint

This pull request introduces a new feature for creating posts via an authenticated API endpoint. It includes changes to the backend logic, API routes, and test cases. Below is a summary of the most important changes grouped by theme:

---

## THE MOST IMPORTANT TO KNOW ABOUT:
- Implements a secure endpoint to create new posts, protected by JWT authentication.
- See Post API Description for an example of how the endpoint works.
- Validates the request body using DTOs and ensures proper error handling.
- All the next text is a description of the PR generated by copilot.

---

## New Feature: Create Posts
- Added a `POST /posts/newPost` endpoint to allow authenticated users to create new posts. The endpoint validates the JWT token, processes the request body, and interacts with the database to store the post.  
  (Files: [`src/features/posts/routes/post.routes.ts`](../src/features/posts/routes/post.routes.ts), [`src/features/posts/controllers/postCrud.controller.ts`](../src/features/posts/controllers/postCrud.controller.ts), [`src/features/posts/services/postCrud.service.ts`](../src/features/posts/services/postCrud.service.ts))

---

## Data Validation and DTOs
- Introduced `createPostSchema` for validating the request body using Yup. Added a `CreatePostDTO` type for strongly-typed data transfer.  
  (File: [`src/features/posts/dto/postCrud.dto.ts`](../src/features/posts/dto/postCrud.dto.ts))

---

## Database Integration
- Implemented repository functions for inserting new posts into the database. These functions handle the SQL logic for persisting post data.  
  (File: [`src/features/posts/repositories/post.crud.repository.ts`](../src/features/posts/repositories/post.crud.repository.ts))

---

## Testing
- Added integration tests for the new controller to validate behavior for authenticated and unauthenticated users, as well as invalid request bodies.  
  (File: [`test-api/controllers/create.post.controller.test.ts`](../test-api/controllers/create.post.controller.test.ts))
- Added repository tests for verifying database insertion behavior and edge cases (e.g., invalid data).  
  (File: [`test-api/repositories/create.post.repository.test.ts`](../test-api/repositories/create.post.repository.test.ts))
- Added route-level tests to ensure the endpoint functions correctly with mocked middleware and controller.  
  (File: [`test-api/routes/create.post.routes.test.ts`](../test-api/routes/create.post.routes.test.ts))

### CODE REVIEW

**Comment by SrDavidC** [comment]
please add a briefly  description about what does your PR.

**Review by angel-chaves — APPROVED** [review]

File: src/features/posts/controllers/postCrud.controller.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      file.mimetype.startsWith('image/') ||
+      file.mimetype === 'image/gif'
+    ) {
+      cb(null, true);
+    } else {
+      cb(null, false);
+      cb(new Error('Solo se permiten archivos de imagen o GIF'));
+    }
+  }
+}).any(); // Permite cualquier campo de archivo
```
Comment (SrDavidC) [inline src/features/posts/controllers/postCrud.controller.ts:23]: Documentation only in english.

File: src/features/posts/dto/postCrud.dto.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    .string()
+    .nullable()
+    .min(1, 'Content must be at least 1 character long')
+    .max(300, 'Content must not exceed 300 characters'),
+
+  imageFile: yup
+    .mixed()
+    .nullable()
+    .test('is-file', 'Image file must be provided', (value) => {
+      // Validar que sea un archivo si está presente
```
Comment (SrDavidC) [inline src/features/posts/dto/postCrud.dto.ts:14]: Documentation only in english.

File: src/features/posts/dto/postCrud.dto.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      return value == null || value instanceof Buffer;
+    }),
+
+  gifFile: yup
+    .mixed()
+    .nullable()
+    .test('is-file', 'GIF file must be provided', (value) => {
+      // Validar que sea un archivo si está presente
```
Comment (SrDavidC) [inline src/features/posts/dto/postCrud.dto.ts:22]: Documentation only in english.

**Review by SrDavidC — CHANGES_REQUESTED** [review]
Check comments.

Good job, nicely done
